src/adapter/src/coord/message_handler.rs

```diff
@@ -343,12 +345,36 @@ impl Coordinator {
                 self.send_peek_response(uuid, response, otel_ctx);
             }
             ControllerResponse::SubscribeResponse(sink_id, response) => {
-                // It's not an error if the subscribe does not exist. When
-                // aborting a subscribe, we eagerly remove it from
-                // `active_subscribes`, but we might receive a few responses
-                // from the controller about the zombie subscribe.
-                if let Some(active_subscribe) = self.active_subscribes.get_mut(&sink_id) {
-                    active_subscribe.process_response(response);
+                match self.active_compute_sinks.get_mut(&sink_id) {
+                    Some(ActiveComputeSink::Subscribe(active_subscribe)) => {
+                        let finished = active_subscribe.process_response(response);
+                        if finished {
+                            self.drop_compute_sinks([(
+                                sink_id,
+                                ComputeSinkRemovalReason::Finished,
+                            )])
+                            .await;
+                        }
+                    }
+                    _ => {
+                        tracing::error!(%sink_id, "received SubscribeResponse for nonexistent subscribe");
+                    }
+                }
+            }
+            ControllerResponse::CopyToResponse(sink_id, response) => {
+                match self.active_compute_sinks.get_mut(&sink_id) {
+                    Some(ActiveComputeSink::CopyTo(active_copy_to)) => {
+                        let response = match response {
+                            Ok(n) => Ok(ExecuteResponse::Copied(usize::cast_from(n))),
+                            Err(error) => Err(AdapterError::Unstructured(error)),
+                        };
+                        active_copy_to.process_response(response);
+                        self.drop_compute_sinks([(sink_id, ComputeSinkRemovalReason::Finished)])
+                            .await;
```

Comment:
@benesch This is after cherry-picking from https://github.com/MaterializeInc/materialize/pull/25260. 

If I remove the entry from `active_compute_sinks` by taking an owned value here (to make the changes for `ctx` to be not an option anymore) and process the response, then `drop_compute_sinks` will not actually clean up the dataflow.

Reply:
One option is, to maybe pass a parameterized `ComputeSinkRemovalReason::Finished`.

Reply:
I'm ok with either 2 or 3 as long as it's really obvious which thing owns retiring the ctx. 3 sounds like it's more explicit so that'd be my first choice.

Reply:
Ack, let me do 3, and folks can weigh in. I think that would be cleaner as well.

Reply:
I'm not sure that'll wind up all that much clearer? Lemme start a discussion on Slack! https://materializeinc.slack.com/archives/C063MF1DAKH/p1708019372268269

Reply:
closing loop from slack https://materializeinc.slack.com/archives/C063MF1DAKH/p1708019372268269, going with 2 here. Do a follow up for a drop_compute_sinks refactor.